src/Shared/Microsoft.DotNet.Kusto/KustoQuery.cs

```diff
@@ -11,28 +11,50 @@ public class KustoQuery
         public KustoQuery()
         {
             Parameters = new List<KustoParameter>();
+            Options = new Dictionary<string, object>();
         }
 
         public KustoQuery(string text, IEnumerable<KustoParameter> parameters)
         {
             Text = text;
             Parameters = parameters.ToList();
+            Options = new Dictionary<string, object>();
         }
 
-        public KustoQuery(string text) : this(text, new List<KustoParameter>())
+        public KustoQuery(string text, Dictionary<string, object> options)
         {
+            Text = text;
+            Parameters = new List<KustoParameter>();
+            Options = options;
+        }
+
+        public KustoQuery(string text, IEnumerable<KustoParameter> parameters, Dictionary<string, object> options)
```

Comment:
It sort of feels like we should only have 1 constructor with a body. Right now we have 4 constructors with bodies, but then 2 that delegate to other constructors.  It seems a little arbitrary on which ones delegate to another and which ones assign the properties themselves.  Granted it was a little arbitrary before, but this almost doubles the number of ctors.

Maybe having the "new Dictionary" be an optional parameter would avoid having to create so many overloads?